src/main/java/com/openlattice/chronicle/services/download/DataDownloadManagerImpl.java

```diff
@@ -0,0 +1,282 @@
+import static com.openlattice.chronicle.util.ChronicleServerUtil.checkNotNullUUIDs;
+import static com.openlattice.edm.EdmConstants.ID_FQN;
+
+/**
+ * @author alfoncenzioka &lt;alfonce@openlattice.com&gt;
+ */
+public class DataDownloadManagerImpl implements DataDownloadManager {
+    protected static final Logger logger = LoggerFactory.getLogger( DataDownloadManagerImpl.class );
+
+    private final CommonTasksManager commonTasksManager;
+
+    public DataDownloadManagerImpl( CommonTasksManager commonTasksManager ) {
+        this.commonTasksManager = commonTasksManager;
+    }
+
+    private Iterable<Map<String, Set<Object>>> getParticipantDataHelper(
+            UUID organizationId,
+            UUID studyId,
+            UUID participantEntityKeyId,
+            UUID srcESID,
+            UUID edgeESID,
+            String token ) {
+
+        try {
+            ApiClient apiClient = new ApiClient( RetrofitFactory.Environment.PROD_INTEGRATION, () -> token );
+            EntitySetsApi entitySetsApi = apiClient.getEntitySetsApi();
+            SearchApi searchApi = apiClient.getSearchApi();
+            EdmApi edmApi = apiClient.getEdmApi();
+
+            /*
+             * 1. get the relevant EntitySets
+             */
+
+            UUID participantESID = commonTasksManager.getParticipantEntitySetId( organizationId, studyId );
+
+            Map<UUID, EntitySet> entitySetsById = entitySetsApi.getEntitySetsById(
+                    ImmutableSet.of( participantESID, srcESID, edgeESID )
+            );
+
+            EntitySet participantsES = entitySetsById.get( participantESID );
+            EntitySet sourceES = entitySetsById.get( srcESID );
+            EntitySet edgeES = entitySetsById.get( edgeESID );
+
+            /*
+             * 2. get all PropertyTypes and set up maps for easy lookups
+             */
+
+            Map<UUID, PropertyType> propertyTypesById = StreamSupport
+                    .stream( edmApi.getPropertyTypes().spliterator(), false )
+                    .collect( Collectors.toMap( PropertyType::getId, Function.identity() ) );
+
+            Map<UUID, Map<UUID, EntitySetPropertyMetadata>> meta =
+                    entitySetsApi.getPropertyMetadataForEntitySets( Set.of( sourceES.getId(), edgeES.getId() ) );
+
+            Map<UUID, EntitySetPropertyMetadata> sourceMeta = meta.get( sourceES.getId() );
+            Map<UUID, EntitySetPropertyMetadata> edgeMeta = meta.get( edgeES.getId() );
+
+            /*
+             * 3. get EntitySet primary keys, which are used later for filtering
+             */
+
+            Set<FullQualifiedName> sourceKeys = edmApi.getEntityType( sourceES.getEntityTypeId() )
+                    .getKey()
+                    .stream()
+                    .map( propertyTypeId -> propertyTypesById.get( propertyTypeId ).getType() )
+                    .collect( Collectors.toSet() );
+
+            Set<FullQualifiedName> edgeKeys = edmApi.getEntityType( edgeES.getEntityTypeId() )
```

Comment:
So, it appears that the property types are used to filter the neighbors returned by filtered neighbor entity search (ie. the fqns in the neighborDetails & associationDetails must be the set of valid fqns for the corresponding entity types) - which is just redundant. This was a carry over from existing logic, and I have no idea why this was implemented that way. Looks safe to delete.

Reply:
Why is that? I can't think of a reason why the key property types specifically should be filtered out as a general rule (unless the goal here is to prevent the ability to write an integration from a downloaded file, which seems unlikely). If there are specific property types that are not desired in the result, it seems safer to manually construct that property type list and reference that list instead of using key property types.

Reply:
I think it would be helpful to get more context here. @UnsungHero97 do you know why we started filtering out primary keys from participant data downloads? ref: https://github.com/openlattice/chronicle-server/commit/596158de14ce0fbdcf68aa48ef626dd77f7cfc4e

Reply:
If I were to guess, it's because the data for the primary key is not necessary in the downloaded file. The EntityType for the raw + processed data EntitySets is `ol.applicationdata` and its primary key is `general.stringid`. Looking at the data stored for `general.stringid`, I see a concatenation of data for several PropertyTypes, for example:

```
APPS-2266com.openlattice.chronicle.debug2020-10-05 18:46:28.171000-07:00
```

`APPS-2266` is the participant id, which is included as part of the file name and is not required to be a column in the data. `com.openlattice.chronicle.debug` is the Android app name, which is already included by another PropertyType. Finally, the timestamp doesn't seem to be actual data, it might just be the time when the `reserveEntityKeyId` was called.

Reply:
Ok, in that case I think it makes sense to manually handcraft the properties to exclude as @katherinebernstein suggested.